src/main/java/com/twitter/joauth/UrlCodec.java

```diff
@@ -105,7 +121,7 @@ public static String normalize(String s) {
       if (c == '%' || c == '+' || c == ',' || c == '[' || c == ']') {
         if (sb == null) {
           sb = new StringBuilder(s.length() + 40); //use length
-          sb.append(s.substring(0, i));
+          sb.append(s, 0, i);
```

Comment:
Overall LGTM! Just curious if we have a way to verify these changes?

Reply:
There is already a test in place: https://github.com/twitter/joauth/blob/master/src/test/scala/com/twitter/joauth/UrlEncoderSpec.scala

I think it's pretty comprehensive but I'm happy to add one or two more cases.

Reply:
nvm~ For some reason I just missed that file. 🚢